app/controllers/TaxYearSummaryController.scala

```diff
@@ -148,45 +149,78 @@ class TaxYearSummaryController @Inject()(taxYearSummaryView: TaxYearSummary,
     }
   }
 
-  private def withTaxYearFinancials(taxYear: Int, isAgent: Boolean)(f: List[DocumentDetailWithDueDate] => Future[Result])
+  private def withTaxYearFinancials(taxYear: Int, isAgent: Boolean)(f: List[TaxYearSummaryChargeItem] => Future[Result])
                                    (implicit user: MtdItUser[_]): Future[Result] = {
 
     financialDetailsService.getFinancialDetails(taxYear, user.nino) flatMap {
       case financialDetails@FinancialDetailsModel(_, documentDetails, _) =>
-        val docDetailsNoPayments = documentDetails.filter(_.paymentLot.isEmpty)
-        val docDetailsCodingOut = docDetailsNoPayments.filter(_.isCodingOutDocumentDetail(isEnabled(CodingOut)))
-        val documentDetailsWithDueDates: List[DocumentDetailWithDueDate] = {
-          docDetailsNoPayments
+
+        def filterMfa(chargeItem: TaxYearSummaryChargeItem): Boolean = {
```

Comment:
If we can move **filterMfa** and **filterReviewAndReconcile** function on the model level?